src/main/java/core/basesyntax/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax.operation;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void execute(Map.Entry<String, Integer> entry, int value) {
+        entry.setValue(entry.getValue() - value);
+    }
```

Comment:
It seems like the `PurchaseOperation` class does not validate if the resulting balance is positive before setting the new value. According to the checklist, you should check the result balance before saving it in the Storage and throw a `RuntimeException` if the balance is negative to avoid negative stock quantities.